api/String.cpp

```diff
@@ -63,18 +63,15 @@ String::String(const __FlashStringHelper *pstr)
 	*this = pstr;
 }
 
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
 String::String(String &&rval)
+	: buffer(rval.buffer)
+	, capacity(rval.capacity)
+	, len(rval.len)
 {
-	init();
-	move(rval);
+	rval.buffer = NULL;
+	rval.capacity = 0;
+	rval.len = 0;
 }
-String::String(StringSumHelper &&rval)
```

Comment:
This was already discussed in the PR in other repo.
Like with pointers and lvalue references, it is possible to convert from derived class pointer or reference to base class pointer or reference. Same goes for rvalue references - Derived&& coverts to Base&&. Because StringSumHelper derives from String, StringSumHelper&& converts to String&& implicitly. Since there is nothing special to be done in case of StringSumHelper there is no need for overload.

Demonstration: https://gcc.godbolt.org/z/zrj64Tao4

Reply:
Right, my bad for not checking the original PR and apparently not even reading the commit message properly, sorry for that. I guess it can indeed be removed and it seems to be adequatly commented in the commit message as well, thanks!